XUtils.h

```diff
@@ -41,26 +41,26 @@ static inline bool String_eq(const char* s1, const char* s2) {
    return strcmp(s1, s2) == 0;
 }
 
-char* String_cat(const char* s1, const char* s2);
+char* String_cat(const char* s1, const char* s2) ATTR_MALLOC;
 
-char* String_trim(const char* in);
+char* String_trim(const char* in) ATTR_MALLOC;
 
-char** String_split(const char* s, char sep, size_t* n);
+char** String_split(const char* s, char sep, size_t* n) ATTR_MALLOC;
 
 void String_freeArray(char** s);
 
-char* String_getToken(const char* line, unsigned short int numMatch);
+char* String_getToken(const char* line, unsigned short int numMatch) ATTR_MALLOC;
 
-char* String_readLine(FILE* fd);
+char* String_readLine(FILE* fd) ATTR_MALLOC;
 
 ATTR_FORMAT(printf, 2, 3)
 int xAsprintf(char** strp, const char* fmt, ...);
 
 ATTR_FORMAT(printf, 3, 4)
 int xSnprintf(char* buf, int len, const char* fmt, ...);
 
-char* xStrdup(const char* str) ATTR_NONNULL;
+char* xStrdup(const char* str) ATTR_NONNULL ATTR_MALLOC;
 
-char* xStrndup(const char* str, size_t len) ATTR_NONNULL;
+char* xStrndup(const char* str, size_t len) ATTR_NONNULL ATTR_MALLOC;
```

Comment:
Maybe `ALLOC_SIZE(2)`?

Reply:
no, we are allocating `size + 1`

Reply:
Would have been to nice, if you could provide a delta to the compiler too …